src/gpu/nvidia/sycl_cuda_compat.cpp

```diff
@@ -0,0 +1,23 @@
+#include "sycl_cuda_compat.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace sycl {
+namespace compat {
+
+template <>
+CUcontext get_native(const ::sycl::device &device) {
+    CUdevice nativeDevice
+            = ::sycl::get_native<::sycl::backend::ext_oneapi_cuda>(device);
+    CUcontext nativeContext;
+    if (cuDevicePrimaryCtxRetain(&nativeContext, nativeDevice)
```

Comment:
Are we expected to release the primary context? What will happen if we continue retaining the primary context without releasing it?

Reply:
Releasing the primary context will cause it to be reset once there are no longer any references to it. I will change this so we release the primary ctx.

Reply:
Change made